### ISSUE TITLE
Create a configuration option to use imagePullSecrets with Couler

### ISSUE THREAD

**Issue by Novak478** [couler-proj/couler#218]
# Summary

Maybe I missed it, but I couldn't seem to find how to configure an image pull secrets within Couler to allow an Argo workflow to pull an image from a private docker repository. I would love to have this as an available configuration option in Couler.  

# Use Cases

I would use this feature in most if not all python/couler apps as our dev, staging, and production repositories are in AWS ECR and require we use ImagePullSecrets. 

# Example Argo workflow using ImagePullSecrets

```yaml
apiVersion: argoproj.io/v1alpha1
kind: workflow
metadata:
  generateName: hello-world
spec:
  entrypoint: whalesay
  imagePullSecrets:
  - name: my-registry-key
  templates:
  - name: whalesay
    container:
      image: "<AWSACCT>.dkr.ecr.us-east-1.amazonaws.com/myimage:latest"
      command: []
      args: []
```
---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by merlintang** [comment]
This is a new feature of Argo. You are welcome to send a Pr @Novak478 :)

**Comment by lcgash** [comment]
supported.

**Comment by terrytangyuan** [comment]
Fixed in #238
